### PR TITLE
[SPARK-50250][SQL] Assign appropriate error condition for `_LEGACY_ERROR_TEMP_2075`: `UNSUPPORTED_FEATURE.WRITE_FOR_BINARY_SOURCE`

### PR DESCRIPTION
**Pull request by itholic**
### What changes were proposed in this pull request?

This PR proposes to Integrate `_LEGACY_ERROR_TEMP_2075 ` into `UNSUPPORTED_FEATURE.WRITE_FOR_BINARY_SOURCE `

### Why are the changes needed?

To improve the error message by assigning proper error condition and SQLSTATE

### Does this PR introduce _any_ user-facing change?

No, only user-facing error message improved

### How was this patch tested?

Updated the existing tests

### Was this patch authored or co-authored using generative AI tooling?

No

### CODE REVIEW

File: common/utils/src/main/resources/error/error-conditions.json [hunk 1]
```diff
@@ -5315,6 +5315,11 @@
         "message" : [
           "Update column nullability for MySQL and MS SQL Server."
         ]
+      },
+      "WRITE_FOR_BINARY_SOURCE" : {
+        "message" : [
+          "Write is not supported for binary file data source."
```
Comment (MaxGekk) [inline common/utils/src/main/resources/error/error-conditions.json:5321]: "is not supported" can be removed. See the full error message:
```
The feature is not supported: Write is not supported for binary file data source.
```
How about:
```suggestion
          "Write for the binary file data source."
```

**Review by MaxGekk — CHANGES_REQUESTED** [review]

File: common/utils/src/main/resources/error/error-conditions.json (hunk 1 above)
Comment (itholic) [reply common/utils/src/main/resources/error/error-conditions.json]: Makes sense. Thanks for the suggestion!

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @itholic.

**Comment by itholic** [comment]
Thanks @MaxGekk for the review

**Pull request closed without merging**